conan/tools/cmake/toolchain/toolchain.py

```diff
@@ -214,8 +215,27 @@ def generate(self):
             else:
                 cache_variables[name] = value
 
+        buildenv, runenv = None, None
+
+        if self._conanfile.conf.get("tools.cmake.cmaketoolchain:presets_environment", default="",
+                                    check_type=str, choices=("disabled", "")) != "disabled":
```

Comment:
Yes, that was intentional, in case in the future we let users generate the environments in the presets with different modes (like for example adding env to configure preset but not to test or inhibiting generating environment in the scripts...)